### PR TITLE
Add auth to navbar, join and leave room if logged in

### PR DESCRIPTION
**Pull request by danieltre23**

### CODE REVIEW

File: ubg/src/pages/Game.js [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
 import { useParams, useHistory } from 'react-router-dom';
-import { useFirestore } from 'reactfire';
+import { useFirestore, AuthCheck } from 'reactfire';
+import Navbar from '../common/Navbar'
 
 export default function Game() {
 
     const { gameId } = useParams();
     const history = useHistory();
+    const [roomId, setRoomId] = useState('')
```
Comment (zhangshyue) [inline ubg/src/pages/Game.js:11]: ;

File: ubg/src/pages/Game.js [hunk 2]
```diff
@@ -1,25 +1,42 @@
-import React from 'react'
+import React, { useState } from 'react'
```
Comment (zhangshyue) [inline ubg/src/pages/Game.js:1]: ;

File: ubg/src/pages/Room.js [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
 import { useParams } from 'react-router-dom';
-import { useFirestore, useFirestoreDocData } from 'reactfire';
-import firebase from 'firebase';
+import { useFirestore, useFirestoreDocData, useUser, useFirestoreCollectionData } from 'reactfire';
 
 export default function Room() {
 
-    const [user, setUser] = useState("Daniel")
-    const { gameId, roomId } = useParams();
+    const [inLobby, setInLobby] = useState(false)
```
Comment (zhangshyue) [inline ubg/src/pages/Room.js:7]: ;

File: ubg/src/pages/Game.js [hunk 4]
```diff
@@ -1,25 +1,42 @@
...
 import AllReviews from '../Reviews/AllReviews';
 import { useParams, useHistory } from 'react-router-dom';
-import { useFirestore } from 'reactfire';
+import { useFirestore, AuthCheck } from 'reactfire';
+import Navbar from '../common/Navbar'
```
Comment (shandyk18) [inline ubg/src/pages/Game.js:5]: Add ;

File: ubg/src/pages/Game.js (hunk 1 above)
Comment (shandyk18) [inline ubg/src/pages/Game.js:11]: Add ;

**Pull request merged**